lib/src/bridge/scanning_mixin.dart

```diff
@@ -22,11 +22,13 @@ mixin ScanningMixin on FlutterBLE {
     List<String> uuids,
     bool allowDuplicates,
   ) {
-    if (_scanEvents == null) {
-      _prepareScanEventsStream();
+    var scanEvents = _scanEvents;
+    if (scanEvents == null) {
+      scanEvents = _prepareScanEventsStream();
+      _scanEvents = scanEvents;
     }
```

Comment:
```suggestion
    var scanEvents = _scanEvents ?? _prepareScanEventsStream();
    _scanEvents = scanEvents;
```

Reply:
Or even better: move the second assignment back to `_prepare...` and instead make it return either existing _scanEvents or the newly created, singleton style

Reply:
Okay, so this is annoying me too, it looks bad but that's a trick to make dart actually leverage its nullcheck static capabilities..
Here is the stupid thing in Dart:
if you just check on `if (_scanEvent == null)` then it wouldn't autounwrap _scanEvent after the line it actually gets a statically non-nullable value on line 27 as you recommended, so even still you'd need to unwrap _scanEvent in the local scope like `foo(_scanEvent!)`which is ugly since it's guaranteed to be nonnull.
The trick is if you do the same but with a `local ` and not with an `object field`. 
like so: `var scanEvents = _scanEvents;`
then no manual unwrap is needed the compiler will know that the local var `scanEvents` has a value for sure while it's unsure about its ivar alias `_scanEvents`.

So to sum up you'll see a lot of boilerplate helper code to mitigate this behaviour of Dart, although it's only 2 extra lines each time, and eliminates the need for `force unwrapping` operators, which is semantically noisy and should be avoided!

Reply:
Yeah, I get it, but if _prepareScanEventsStream() always returns non-null and it does the 
```dart
if (_scanEvents == null) { 
   // create a new stream and assign it to _scanEvents
 }
 
 return _scanEvents
 ```
 
you still get that behaviour, but you get rid of the assignment in a function that does not care about it.

Reply:
it will have to be `return _scanEvents!` actually, see below, I'll explain more thoroughly at the point of usage of scanEvenets.

Reply:
I don't mind a helper var in `_prepareScanEventsStream()`, I just don't see the point of having a function return non-nullable if we still need to check for null and reassign in the consumer of said function.

Reply:
Okay gotcha, now _scanEvents is a lazy getter, and _prepareScanEventsStream is gone, this way caller don't need to check anything. I actually would have written it like this myself, just again didn't want to introduce too much change out of the scope of this PR ;)